v-next/core/src/internal/tasks/resolved-task.ts

```diff
@@ -69,24 +77,190 @@ export class ResolvedTask implements Task {
+   * it doesn't have a default value.
+   *
+   * @throws HardhatError if the parameter is required and doesn't have a value.
+   */
+  #validateRequiredParameter(
+    parameter: TaskParameter,
+    value: ParameterValue | ParameterValue[],
+  ) {
+    if (parameter.defaultValue === undefined && value === undefined) {
+      throw new HardhatError(
+        HardhatError.ERRORS.ARGUMENTS.MISSING_VALUE_FOR_PARAMETER,
+        {
+          paramName: parameter.name,
+        },
+      );
+    }
+  }
+
+  /**
+   * Validates that a parameter has the correct type. If the parameter is optional
+   * and doesn't have a value, the type is not validated as it will be resolved
+   * to the default value.
+   *
+   * @throws HardhatError if the parameter has an invalid type.
+   */
+  #validateParameterType(
+    parameter: TaskParameter,
+    value: ParameterValue | ParameterValue[],
+  ) {
+    // skip type validation for optional parameters with undefined value
+    if (value === undefined && parameter.defaultValue !== undefined) {
+      return;
+    }
+
+    // check if the parameter is variadic
+    const isPositionalParameter = (
+      param: TaskParameter,
+    ): param is PositionalTaskParameter => "isVariadic" in param;
+    const isVariadic = isPositionalParameter(parameter) && parameter.isVariadic;
+
+    // check if the value is valid for the parameter type
+    if (!isParameterValueValid(parameter.parameterType, value, isVariadic)) {
+      throw new HardhatError(
+        HardhatError.ERRORS.ARGUMENTS.INVALID_VALUE_FOR_TYPE,
+        {
+          value,
+          name: parameter.name,
+          type: parameter.parameterType,
+        },
+      );
+    }
+  }
+
+  /**
+   * Validates that no extra arguments were provided in the task arguments.
+   *
+   * @throws HardhatError if extra arguments were provided. The error message
+   * includes the name of the first extra argument.
+   */
+  #validateExtraArguments(providedArgumentNames: Set<string>) {
+    if (providedArgumentNames.size > 0) {
+      throw new HardhatError(
+        HardhatError.ERRORS.ARGUMENTS.UNRECOGNIZED_NAMED_PARAM,
+        {
+          parameter: [...providedArgumentNames][0],
+        },
+      );
+    }
+  }
+
+  /**
+   * Resolves the action file for a task. The action file is imported and the
+   * default export function is returned.
+   *
+   * @throws HardhatError if the module can't be imported or doesn't have a
+   * default export function.
+   */
+  async #resolveFileAction(
+    actionFileUrl: string,
+    taskId: string[],
+  ): Promise<NewTaskActionFunction | TaskOverrideActionFunction> {
+    let resolvedActionFn;
+    try {
+      resolvedActionFn = await import(actionFileUrl);
+    } catch (error) {
+      ensureError(error);
+      throw new HardhatError(
+        HardhatError.ERRORS.TASK_DEFINITIONS.INVALID_ACTION_URL,
+        {
+          action: actionFileUrl,
+          task: formatTaskId(taskId),
+        },
+        error,
+      );
```

Comment:
To be addressed in a separate PR, but here is where you need to "validate" the plugin's installation.

Reply:
Yes, I've added it to the main task issue description.